src/native/eventpipe/ep.c

```diff
@@ -246,8 +246,10 @@ ep_provider_callback_data_alloc_copy (EventPipeProviderCallbackData *provider_ca
 	EventPipeProviderCallbackData *instance = ep_rt_object_alloc (EventPipeProviderCallbackData);
 	ep_raise_error_if_nok (instance != NULL);
 
-	if (provider_callback_data_src)
+	if (provider_callback_data_src != NULL) {
```

Comment:
nit: same meaning as original expression, in C we normally just do if checks on pointers, like if (x) or if (!x), no need to compare against NULL.